jdk/src/share/classes/java/util/ServiceLoader.java

```diff
@@ -375,7 +375,11 @@ private boolean hasNextService() {
                                 provNames.add(className);
                             }
                         } catch (ClassNotFoundException x) {
-                            // If class not found, then don't add it.
+                            /* If class cannot found, there is no way of knowing if it is a subclass.
+                             * of java.security.Provider, so allow it anyway and it will be caught
```

Comment:
Please improve this comment; I think it's missing "be" in "cannot be found" and there shouldn't be a period in the middle of the sentence. Please shorten these lines (> 100 characters is too wide).

Reply:
Sorry about that. Fixed.